### DISCUSSION TITLE
Is there a way to wait for the async call using Composition API that would block the page rendering?

### DISCUSSION THREAD

**Discussion by DamianGlowala** [nuxt/nuxt#10212] [Questions (Nuxt 2)] [answered]
Hey Nuxters!

Does any of you know if there is a way to use Composition API in **Nuxt 2** with an asynchronous call that would block the execution until the call is resolved? I have a use-case when I would either load the page with the data being successfully downloaded and present from the very first moment OR redirect to somewhere else (without using any loading states, etc.). Can't see anything that would prevent the `setup()` function from waiting for the call to resolve.

Appreciate any advice :)

**Comment by DamianGlowala** [comment, marked as answer]
@danielroe Would you be kind to shed some light on the above matter? I am currently stuck with no solution other than displaying a loading state in the UI 😢 Is there any way to fetch data asynchronously with Composition API and block the routing until resolved?

**Reply by danielroe** [reply]
There is no way to block the page loading on client side navigation in Vue 2/Nuxt 2 composition API. At the moment your only choice is to use asyncData with options API (for that piece) or display a loading state.
